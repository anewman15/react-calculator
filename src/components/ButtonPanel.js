/*
 eslint-disable no-unused-vars
*/

import React from 'react';
import Button from './Button';

const ButtonPanel = props => {
  const btnGroups = [
    ['AC', '+/-', '%', '/'],
    ['7', '8', '9', 'X'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '='],
  ];

  const handleClick = props.clickHandler;

  const keyButtons = btnGroups.map(group => (
    <div key={group.toString()}>
      {group.map(name => (
        <Button key={name.toString()} btnName={name} clickHandler={handleClick}/>
      ))}
    </div>
  ));

  return (
    <div>
      {keyButtons}
    </div>
  );
};

export default ButtonPanel;