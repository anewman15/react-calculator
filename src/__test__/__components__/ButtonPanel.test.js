/*
 eslint-disable no-unused-vars, import/named, import/no-cycle
*/

import React from 'react';
import { act, render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import pretty from 'pretty';
import ButtonPanel from '../../components/ButtonPanel';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('ButtonPanel', () => {
  it('should render groups of buttons', () => {
    const tree = renderer
      .create(<ButtonPanel />)
      .toJSON();

    expect(tree)
      .toMatchSnapshot();
  });
});
