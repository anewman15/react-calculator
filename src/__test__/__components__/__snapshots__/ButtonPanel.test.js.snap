// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel should render groups of buttons 1`] = `
<div
  className="btn-panel"
>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-clear"
      onClick={[Function]}
      style={null}
    >
      AC
    </button>
    <button
      className="btn btn-toggle"
      onClick={[Function]}
      style={null}
    >
      +/-
    </button>
    <button
      className="btn btn-toggle"
      onClick={[Function]}
      style={null}
    >
      %
    </button>
    <button
      className="btn btn-operator"
      onClick={[Function]}
      style={null}
    >
      /
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      7
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      8
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      9
    </button>
    <button
      className="btn btn-operator"
      onClick={[Function]}
      style={null}
    >
      X
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      4
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      5
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      6
    </button>
    <button
      className="btn btn-operator"
      onClick={[Function]}
      style={null}
    >
      -
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      1
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      2
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      3
    </button>
    <button
      className="btn btn-operator"
      onClick={[Function]}
      style={null}
    >
      +
    </button>
  </div>
  <div
    className="btn-group"
  >
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={
        Object {
          "width": "50%",
        }
      }
    >
      0
    </button>
    <button
      className="btn btn-digit"
      onClick={[Function]}
      style={null}
    >
      .
    </button>
    <button
      className="btn btn-equal"
      onClick={[Function]}
      style={null}
    >
      =
    </button>
  </div>
</div>
`;
